spec_42.txt

```diff
@@ -0,0 +1,179 @@
+
+* To discuss a patch, people MAY comment on the Platform pull request, on the commit, or elsewhere.
+
+* To accept or reject a patch, a Maintainer SHALL use the Platform interface.
+
+* Maintainers SHOULD NOT merge their own patches except in exceptional cases, such as non-responsiveness from other Maintainers for an extended period (more than 1-2 days).
+
+* Maintainers SHALL NOT make value judgments on correct patches.
+
+* Maintainers SHALL merge correct patches from other Contributors rapidly.
+
+* Maintainers MAY merge incorrect patches from other Contributors with the goals of (a) ending fruitless discussions, (b) capturing toxic patches in the historical record, (c) engaging with the Contributor on improving their patch quality.
+
+* The Contributor MAY tag a user's issue as "Ready" after making a pull request for the issue.
+
+* The user who created an issue SHOULD close the issue after checking the patch is successful.
+
+* Any Contributor who has value judgments on a patch SHOULD express these via their own patches.
+
+* Maintainers MAY commit changes to non-source documentation directly to the project.
+
++++ Branches and Releases
+
+* The project SHALL have one branch ("master") that always holds the latest in-progress version and SHOULD always build.
+
+* The project SHALL NOT use topic branches for any reason. Personal forks MAY use topic branches.
+
+* To make a stable release a Maintainer shall tag the repository. Stable releases SHALL always be released from the repository master.
+
++++ Evolution of Public Contracts
+
+* All Public Contracts (APIs or protocols) SHALL be documented.
+
+* All Public Contracts SHOULD have space for extensibility and experimentation.
+
+* A patch that modifies a stable Public Contract SHOULD not break existing applications unless there is overriding consensus on the value of doing this.
+
+* A patch that introduces new features SHOULD do so using new names (a new contract).
+
+* New contracts SHOULD be marked as "draft" until they are stable and used by real users.
+
+* Old contracts SHOULD be deprecated in a systematic fashion by marking them as "deprecated" and replacing them with new contracts as needed.
+
+* When sufficient time has passed, old deprecated contracts SHOULD be removed.
+
+* Old names SHALL NOT be reused by new contracts.
+
++++ Project Administration
+
+* The project founders SHALL act as Administrators to manage the set of project Maintainers.
+
+* The Administrators SHALL ensure their own succession over time by promoting the most effective Maintainers.
+
+* A new Contributor who makes correct patches, who clearly understands the project goals, and the process SHOULD be invited to become a Maintainer.
+
+* Administrators SHOULD remove Maintainers who are inactive for an extended period of time, or who repeatedly fail to apply this process accurately.
```

Comment:
Admins, not maintainers, do this cleanup. We at least need a rule to eject
bad maintainers.

On Thu, May 5, 2016 at 1:51 PM, Kevin Sapper <notifications@github.com>
wrote:

> In spec_42.txt
> <https://github.com/zeromq/rfc/pull/84#discussion_r62177455>:
>
> > +
> > +* Old contracts SHOULD be deprecated in a systematic fashion by marking them as "deprecated" and replacing them with new contracts as needed.
> > +
> > +* When sufficient time has passed, old deprecated contracts SHOULD be removed.
> > +
> > +* Old names SHALL NOT be reused by new contracts.
> > +
> > ++++ Project Administration
> > +
> > +* The project founders SHALL act as Administrators to manage the set of project Maintainers.
> > +
> > +* The Administrators SHALL ensure their own succession over time by promoting the most effective Maintainers.
> > +
> > +* A new Contributor who makes correct patches, who clearly understands the project goals, and the process SHOULD be invited to become a Maintainer.
> > +
> > +* Administrators SHOULD remove Maintainers who are inactive for an extended period of time, or who repeatedly fail to apply this process accurately.
>
> Did we ever apply this? This may put Maintainers in a uncomfortable
> situation and maybe enable bad actors to take control.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/rfc/pull/84/files/9d3e8549a9d9c5fb24a563175b6890a26dbd9581#r62177455>
>